tests/Mutator/Regex/PregMatchMatchesTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Mutator\Regex;
+
+use Infection\Tests\Mutator\AbstractMutatorTestCase;
+
+/**
+ * @internal
+ */
+final class PregMatchMatchesTest extends AbstractMutatorTestCase
+{
+    /**
+     * @dataProvider providesMutatorCases
+     *
+     * @param string $input
+     * @param string|null $output
+     */
+    public function test_mutator(string $input, string $output = null)
+    {
+        $this->doTest($input, $output);
+    }
+
+    public function providesMutatorCases(): \Generator
+    {
+        yield 'It mutates ' => [
+            <<<'PHP'
+<?php
+
+preg_match('/a/', 'b', $foo);
+PHP
+            ,
+            <<<'PHP'
+<?php
+
+$foo = array();
+PHP
+        ];
+
+        yield 'It does not mutate if the function is a variable' => [
+            <<<'PHP'
+<?php
+
+$foo = 'preg_match';
+$foo('/a/', 'b', $bar);
+PHP
+        ];
+        yield 'It mutates if preg_match is incorrectly cased' => [
+          <<<'PHP'
+<?php
+
+PreG_maTch('/a/', 'b', $foo);
+PHP
+            ,
+            <<<'PHP'
+<?php
+
+$foo = array();
+PHP
+        ];
+        yield 'It does not mutate if there are less than 3 arguments' => [
```

Comment:
Shouldn't there be a test case with more than 3 args? preg_match can have as much as 5 args.